decorator.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
```

Comment:
- [ ] In the constructor kindly assign a nameable object from params to an instance variable as stated in the project requirement 🙏